com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/dependency/CopybookDependencyServiceImpl.java

```diff
@@ -40,6 +36,46 @@
   private static final String DEP_EXTENSION = ".dep";
   @Getter private List<Path> workspaceFolderPaths;
 
+  /**
+   * This method is invoked to handle the insertion of a copybook name into the dependency file
+   *
+   * @param event contains information the copybook (document URI and event sync type)
+   * @param requiredCopybookName name that represent the new copybook that is supposed to be written
+   * @param targetPaths the reference path where create or search a dependency file.
+   */
+  @Override
+  public void addCopybookInDepFile(
+      RequiredCopybookEvent event, String requiredCopybookName, List<Path> targetPaths) {
+    setWorkspaceFolderPaths(targetPaths);
+    if (isFileInDidOpen(event) || isNestedCopybookBeProcessed(event)) {
+      writeCopybookInDepFile(requiredCopybookName, event.getDocumentUri());
+    }
+  }
+
+  private boolean isNestedCopybookBeProcessed(RequiredCopybookEvent event) {
+    return isDocumentInDidOpen(event)
+        && getExtensionFromURI(event.getDocumentUri()).equalsIgnoreCase("cpy");
+  }
+
+  private boolean isDocumentInDidOpen(RequiredCopybookEvent event) {
```

Comment:
why are there two methods which are doing most likely the same thing? `isDocumentInDidOpen` and `isFileInDidOpen`

Reply:
cause I missed it 👍

Reply:
🦄